### ISSUE TITLE
There is no clear document for how `require "foo/bar/baz"` work.

### ISSUE THREAD

**Issue by zw963** [crystal-lang/crystal-book#677]
After i test it, it seem like the lookup path is: (assume CRYSTAL_PATH is lib)

```
      lib/foo/bar/baz.cr
      lib/foo/bar/baz/baz.cr
      lib/foo/src/bar/baz.cr
```

But, i never saw any document mention it.

**Comment by straight-shoota** [comment]
Here you go: https://crystal-lang.org/reference/1.7/syntax_and_semantics/requiring_files.html#other-forms

**Comment by zw963** [comment]
> Here you go: https://crystal-lang.org/reference/1.7/syntax_and_semantics/requiring_files.html#other-forms I think that should be sufficient documentation?

Oops, sorry, i don't know why i missing it!

BTW: the explain in other-forms is not accurate, because `require "foo/bar/baz"` seem like never search `foo/src/bar/baz/baz.cr`.

**Comment by straight-shoota** [comment]
Indeed that seems to be the case. I figure this might be an unintended omission. So we probably need to change the implementation, not the documentation.

I think this might've eventually been lost in https://github.com/crystal-lang/crystal/issues/10820.

I created https://github.com/crystal-lang/crystal/issues/13210 to address the issues with require paths.

**Comment by zw963** [comment]
Oops, i thought i see it before, but i lose completely now for how `require` search file.

Let us not think about our implementation(it probably wrong, right?)

What is the expected search path for following requires? (for lib only)

```crystal
require "foo"

# lib/foo.cr
# lib/foo/foo.cr
# lib/foo/src/foo.cr
# ???
```

```crystal
require "foo/bar/baz"

# lib/foo/bar/baz.cr
# lib/foo/bar/baz/baz.cr
# lib/foo/src/bar/baz.cr
# ???
```

I remember when i first use Crystal (1.4), i read document somewhere said `require "foo"` will search `lib/foo/src/foo/foo.cr`,
i even try it, it works. but, obviously, it not work anymore now for 1.7.2.

**Comment by zw963** [comment]
Oops, i thought i see it before, but i lose completely now for how `require` search file.

Let us not think about our implementation(it probably wrong, right?)

What is the expected search path for following requires? (for lib only)

```crystal
require "foo"

# lib/foo.cr
# lib/foo/foo.cr
# lib/foo/src/foo.cr
# ???
```

```crystal
require "foo/bar/baz"

# lib/foo/bar/baz.cr
# lib/foo/bar/baz/baz.cr
# lib/foo/src/bar/baz.cr
# ???
```

I remember when i first use Crystal (1.4), i read document somewhere said `require "foo"` will search `lib/foo/src/foo/foo.cr`,
i even try it, it works. but, obviously, it not work anymore now for 1.7.2.

**Comment by straight-shoota** [comment]
No, `lib/foo/src/foo/foo.cr` was ever looked up, neither in Crystal 1.0 nor 1.4. I checked that.
